tpe/plugin/src/SDFFeatures.cc

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <sdf/Box.hh>
+#include <sdf/Cylinder.hh>
+#include <sdf/Mesh.hh>
+#include <sdf/Sphere.hh>
+#include <sdf/Geometry.hh>
+#include <ignition/common/Console.hh>
+
+#include "SDFFeatures.hh"
+
+using namespace ignition;
+using namespace physics;
+using namespace tpeplugin;
+
+/////////////////////////////////////////////////
+Identity SDFFeatures::ConstructSdfWorld(
+    const Identity &_engine,
+    const ::sdf::World &_sdfWorld)
+{
+  const Identity worldID = this->ConstructEmptyWorld(_engine, _sdfWorld.Name());
+
+  // construct models
+  for (std::size_t i = 0; i < _sdfWorld.ModelCount(); ++i)
+  {
+    this->ConstructSdfModel(worldID, *_sdfWorld.ModelByIndex(i));
+  }
+
+  return worldID;
+}
+
+/////////////////////////////////////////////////
+Identity SDFFeatures::ConstructSdfModel(
+  const Identity &_worldID,
+  const ::sdf::Model &_sdfModel)
+{
+  // Read sdf params
+  const std::string name = _sdfModel.Name();
+  igndbg << "construct model " << name << std::endl;
```

Comment:
I think I may have added these. We can remove them now that there tests verifying the entities are created.